tuf/ngclient/_internal/requests_fetcher.py

```diff
@@ -99,26 +99,12 @@ def _chunks(self, response: "requests.Response") -> Iterator[bytes]:
         download."""
 
         try:
-            while True:
-                # We download a fixed chunk of data in every round. This is
-                # so that we can defend against slow retrieval attacks.
-                # Furthermore, we do not wish to download an extremely
-                # large file in one shot.
-
-                # NOTE: This may not handle some servers adding a
-                # Content-Encoding header, which may cause urllib3 to
-                #  misbehave:
-                # https://github.com/pypa/pip/blob/404838abcca467648180b358598c597b74d568c9/src/pip/_internal/download.py#L547-L582
-                data = response.raw.read(self.chunk_size)
-
-                # We might have no more data to read, we signal
-                # that the download is complete.
-                if not data:
-                    break
-
+            for data in response.iter_content(self.chunk_size):
```

Comment:
Could you remind me why we used to read from `response.raw`,  why it is not necessary anymore, and how it is related to compression?

Reply:
Do we give up any needed control about the incoming data stream?

Reply:
oh, great question, should have talked about that.

I believe this was added to pip for this case:
* broken servers might slap "Content-encoding: gzip" when they serve a file that happens to be compressed: they might do this even if client did not ask for compression
* if response includes "Content-encoding: gzip", requests will decompress (even if explicitly did not request compression)
* the application can workaround this by reading(non-decompressed) response.raw but obviously you can only do that if you 100% know that you did not request compression -- otherwise it makes no sense

Reply:
Thanks for the insight. Would you mind adding a short note, about that to the commit message, where you say "Don't use response.raw in RequestsFetcher"